qualtran/bloqs/qsp/generalized_qsp.py

```diff
@@ -214,6 +202,44 @@ def safe_angle(x):
     return theta, phi, lambd
 
 
+def scale_down_to_qsp_polynomial(
+    P: Sequence[complex], *, n_points: int = 10**5
+) -> NDArray[np.complex_]:
+    r"""Scale down the polynomial to be a valid QSP Polynomial
+
+    $P$ is a QSP polynomial if $|P(e^{i\theta})| \le 1$ for every $\theta \in [0, 2\pi]$.
+
+    If the input polynomial is not a valid QSP polynomial, this function attempts to compute
+    the maximum absolute value on the unit circle, and scale it down by that factor.
+    Otherwise returns the input as-is.
```

Comment:
Add a link to the open issue where we are discussing potential improvements to this method.